### PR TITLE
resize update

### PR DESCRIPTION
**Pull request by sugi-chan**
@gsganden 

defaulting to None with optional int and tuples. added a check to the save response as png function to check for int or tuple else it defaults to original size like before.

Added resize on download to cut out a step where images would need to be read back in and resized before final use. 

Pull Request Checklist
 - [X ] Pull request is based on WIP branch
 - [X ] Pull request includes a description of the change and the reason behind it
 - [ ] `pytest` passes locally: works with make_dataset and download_images files
 - [X ] `flake8 creevet tests` passes locally
 - [X ] CHANGELOG has been updated
 - [ X] Version in `_version.py` has been updated

So the code works when I test with hedwig files. but running one of the tests on download.py I was not able to download a file... but it looks like a 404 error? Below are the parameters of that run. The resize_shape defaults to None as it should. 

url = 'https://vignette.wikia.nocookie.net/p__/images/7/73/Mad-Eye_Moody%27s_%22mad_eye%22.jpeg/revision/latest?path-prefix=protagonistxyz'
path_func = functools.partial(<function replace_directory_and_extension at 0x117a591e0>, extension='.png', outdir='/Users/msugimura/Documents/GitHub/creevey/tests/sample_data')
write_func = functools.partial(<function write_response_as_png at 0x117a59268>, write_log_path=None, resize_shape=None), skip_existing = True

Error message:
DEBUG:urllib3.connectionpool:https://vignette.wikia.nocookie.net:443 "GET /smurfsfanon/images/3/32/Philosopher%27s_Stone.jpgxyz HTTP/1.1" 404 1328
ERROR:root:Failed to download https://vignette.wikia.nocookie.net/smurfsfanon/images/3/32/Philosopher%27s_Stone.jpgxyz with status code 404

### CODE REVIEW

**Comment by gsganden** [comment]
Weird, the tests are working fine for me, including on your fork.

File: creevey/download.py [hunk 1]
```diff
@@ -289,8 +301,12 @@ def _initialize_write_log(path):
         write_log.write('timestamp,url,local_path\n')
 
 
-def _save_response_content_as_png(response, path):
+def _save_response_content_as_png(response, path, resize_shape: Optional[Union[int, Tuple[int, int]]] = None):
     image = Image.open(io.BytesIO(response.content))
+    if isinstance(resize_shape, tuple) is True:
```
Comment (gsganden) [inline creevey/download.py]: Remove `is True` -- it isn't necessary, and PEP 8 recommends not including it.

File: creevey/download.py [hunk 2]
```diff
@@ -289,8 +301,12 @@ def _initialize_write_log(path):
...
+        image = image.resize((resize_shape[0], resize_shape[1]))
+    if isinstance(resize_shape, int) is True:
```
Comment (gsganden) [inline creevey/download.py]: Remove `is True`.

**Comment by sugi-chan** [comment]
It is probably some weird issue on my end then.

File: creevey/download.py (hunk 1 above)
Comment (gsganden) [inline creevey/download.py]: Another approach that I think I slightly prefer is to turn `resize_shape` into a `tuple` right off the bat:

```
if isinstance(resize_shape, int):
    resize_shape = (resize_shape, resize_shape)
image = image.resize(resize_shape)
```

File: creevey/download.py [hunk 3]
```diff
@@ -60,7 +66,7 @@ def download_images_as_png(urls: Iterable[str],
     if outdir is not None:
         path_func = partial(replace_directory_and_extension, extension='.png', outdir=outdir)
 
-    write_func = partial(write_response_as_png, write_log_path=write_log_path)
+    write_func = partial(write_response_as_png, write_log_path=write_log_path, resize_shape=resize_shape)
```
Comment (gsganden) [inline creevey/download.py]: I would stack the arguments here.

File: creevey/download.py [hunk 4]
```diff
@@ -48,6 +49,11 @@ def download_images_as_png(urls: Iterable[str],
         Desired path for logs of successful writes
     n_jobs
         Number of jobs to run in parallel
+    resize_shape: int or tuple
+        Desired final image dimensions to resize to. Default is None
+        which will not change image size on download. a single int
+        will make a square of that size. a tuple will create a custome
+        x,y rectangle.
```
[the remark this answers is not in the document]
Comment (sugi-chan) [reply creevey/download.py]: makes sense, will do!

[the remark this answers is not in the document]
Comment (sugi-chan) [reply creevey/download.py]: the order on the pil image resize is (width, height) so I just have to swap the order the values that come out of the tuple. easy to do

File: creevey/download.py (hunk 1 above)
Comment (gsganden) [reply creevey/download.py]: Yes

File: creevey/download.py [hunk 5]
```diff
@@ -301,12 +303,14 @@ def _initialize_write_log(path):
...
+                                  resize_shape: Optional[Union[int, Tuple[int, int]]] = None):
     image = Image.open(io.BytesIO(response.content))
-    if isinstance(resize_shape, tuple) is True:
-        image = image.resize((resize_shape[0], resize_shape[1]))
-    if isinstance(resize_shape, int) is True:
-        image = image.resize((resize_shape, resize_shape))
+    if resize_shape is not None:
+        if isinstance(resize_shape, int):
+            resize_shape = (resize_shape, resize_shape)
+        image = image.resize((resize_shape[1], resize_shape[0]))
```
Comment (gsganden) [inline creevey/download.py]: Let's make it slightly more self-documenting:

```
height, width = resize_shape
image = image.resize((width, height))
```

Comment (sugi-chan) [reply creevey/download.py]: good change!

**Pull request merged**